src/js/common/validate.js

```diff
@@ -0,0 +1,192 @@
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.schema.validator");
+
+gpii.schema.validator.validate = function (that, key, content) {
+    if (!that.schemaContents[key]) {
+        fluid.fail("You tried to validate a component for which we have no schema content.  Can't continue.");
+    }
+
+    // We instantiate a new validator each time to avoid detecting errors from previous runs or from other sessions.
+    var validator = that.getValidator();
+
+    // We have to validate all schemas at once to a) confirm that we have usable schemas and b) handle dependencies between schemas correctly.
+    var schemasValid = validator.validateSchema(Object.keys(that.schemaContents).map(function (v) { return that.schemaContents[v]; }));
+    if (!schemasValid) {
+        fluid.fail(validator.getLastErrors());
+    }
+
+    var contentValid = validator.validate(content, that.schemaContents[key]);
+    if (!contentValid) {
+        return (gpii.schema.validator.sanitizeValidationErrors(validator.getLastErrors()));
+    }
+
+    return undefined;
+};
+
+/*
+ z-schema gives us output like:
+
+ [
+     {
+         "code": "OBJECT_MISSING_REQUIRED_PROPERTY",
+         "params":["termLabel"],
+         "message":"Missing required property: termLabel",
+         "path":"#/"
+     },
+     {
+         "code":"PATTERN",
+         "params":["^[a-z]+([A-Z][a-z]+)*$","6DotComputerBrailleTable"],
+         "message":"String does not match pattern ^[a-z]+([A-Z][a-z]+)*$: 6DotComputerBrailleTable",
+         "path":"#/uniqueId"
+     }
+ ]
+
+ See https://github.com/zaggino/z-schema/blob/master/src/Errors.js for the list of errors and
+ https://github.com/zaggino/z-schema/blob/master/src/JsonValidation.js for the logic behind them.
+
+ We need to turn this into something human-readable, especially for pattern-based matches.
+
+ We also need to break it down by field so that we can show feedback in-context;
+
+ */
+gpii.schema.validator.sanitizeValidationErrors = function (errors) {
+    var sanitizedErrors = { documentErrors: [], fieldErrors: {}};
+
+    fluid.each(errors, function (error) {
+        gpii.schema.validator.sanitizeError(error, sanitizedErrors);
+    });
+
+    return sanitizedErrors;
+};
+
+gpii.schema.validator.sanitizeError = function (error, errorMap) {
+    // Errors with fields that contain data are already associated with the field based on the path
+    var path = gpii.schema.validator.extractPathSegments(error);
+
+    gpii.schema.validator.saveToPath(path, error.message, errorMap);
+};
+
+// Filter to remove empty array segments.  Required because strings like `#/` would otherwise result in an
+// extra array element when we run them through `split`.
+gpii.schema.validator.removeEmptySegments = function (el) { return el !== undefined && el !== null && el.length > 0; };
+
+/*
+  Z-Schema represents validation failures using path notation like `#/category/subcategory/field`.  This function
```

Comment:
If the field name contains a slash and is missing, its name is left intact (and found in params[0]).

If the field is invalid or is missing sub-content, it will be reported in the path, and is escaped, as in:

```
[
  {
    "code": "OBJECT_MISSING_REQUIRED_PROPERTY",
    "params": [
      "required"
    ],
    "message": "Missing required property: required",
    "path": "#/deep~1slasher"
  }
]
```
I can add handling and tests for the special case shortly, as well as a comment where the replace takes place.